l10n_co_res_partner/models/l10n_co_res_partner.py

```diff
@@ -0,0 +1,459 @@
+    def _concat_name(self):
+        """
+        This function concatenates the four name fields in order to be able to
+        search for the entire name. On the other hand the original name field
+        should not be editable anymore as the new name fields should fill it up
+        automatically.
+        @return: void
+        """
+        # Avoiding that "False" will be written into the name field
+        if self.name1 is False:
+            self.name1 = ''
+
+        if self.name2 is False:
+            self.name2 = ''
+
+        if self.lastname1 is False:
+            self.lastname1 = ''
+
+        if self.lastname2 is False:
+            self.lastname2 = ''
+
+        # Collecting all names in a field that will be concatenated
+        nameList = [
+            self.name1.encode(encoding='utf-8').strip(),
+            self.name2.encode(encoding='utf-8').strip(),
+            self.lastname1.encode(encoding='utf-8').strip(),
+            self.lastname2.encode(encoding='utf-8').strip()
+        ]
+
+        formatedList = []
+        if self.companyName is False:
+            if self.type == 'delivery':
+                self.name = self.pos_name
+                self.name1 = False
+                self.name2 = False
+                self.lastname1 = False
+                self.lastname2 = False
+                self.doctype = 1
+            else:
+                for item in nameList:
+                    if item is not '':
+                        formatedList.append(item)
+                self.name = ' ' .join(formatedList)
+        else:
+            self.name = self.companyName
+
+    @api.onchange('name')
+    def onChangeName(self):
+        """
+        The name field gets concatenated by the four name fields.
+        If a user enters a value anyway, the value will be deleted except first
+        name has no value. Reason: In certain forms of odoo it is still
+        possible to add value to the original name field. Therefore we have to
+        ensure that this field can receive values unless we offer the four name
+        fields.
+        @return: void
+        """
+        if self.name1 is not False:
+            if len(self.name1) > 0:
+                self._concat_name()
+        if self.companyName is not False:
+            if len(self.companyName) > 0:
+                self._concat_name()
+
+    @api.onchange('personType')
+    def onChangePersonType(self):
+        """
+        Delete entries in name and company fields once the type of person
+        changes. This avoids unnecessary entries in the database and makes the
+        contact cleaner and ready for analysis
+        @return: void
+        """
+        if self.personType is 2:
+            self.name1 = ''
+            self.name2 = ''
+            self.lastname1 = ''
+            self.lastname2 = ''
+            self.pn_retri = 7
+        elif self.personType is 1:
+            self.companyName = False
+            self.pn_retri = False
```

Comment:
Reading the reason, I think we will need a report (SQL view) instead create computed concatenated fields ar cleanuup of this information.

The proper way is do as Account does in the report folder you will find some good examples.